### ISSUE TITLE
Evaluate the dice score.

### ISSUE THREAD

**Issue by qimingyudaowenti** [milesial/Pytorch-UNet#288]
After `sigmoid`, the threshold is 0.5, so should it be:
 ```python
mask_pred = (F.sigmoid(mask_pred) > 0.5).float()
```
instead of `>0`:
https://github.com/milesial/Pytorch-UNet/blob/5a7a0d5a57467e43e8d10c9b28a99b67e0936803/evaluate.py#L27

I am also confused about this line:
https://github.com/milesial/Pytorch-UNet/blob/5a7a0d5a57467e43e8d10c9b28a99b67e0936803/evaluate.py#L32
The label of background may be 0 in most segmentation task, this line of code seems convert the background to 1 and foreground to 0, test as below:
```python
import torch
from torch.nn.functional import one_hot

img = torch.arange(0, 18).view(2, 3, 3)
mask = img % 2
label_one_hot = one_hot(mask)
b = label_one_hot.permute(0, 3, 1, 2)
background = b[:, :1, ...]
print(f'{mask=}')
print(f'{background=}')
```
output:
```
mask=tensor([[[0, 1, 0],
         [1, 0, 1],
         [0, 1, 0]],

        [[1, 0, 1],
         [0, 1, 0],
         [1, 0, 1]]])
background=tensor([[[[1, 0, 1],
          [0, 1, 0],
          [1, 0, 1]]],


        [[[0, 1, 0],
          [1, 0, 1],
          [0, 1, 0]]]])
```
So [dice_coeff](https://github.com/milesial/Pytorch-UNet/blob/5a7a0d5a57467e43e8d10c9b28a99b67e0936803/utils/dice_score.py#L5) function may give a wrong result?

**Comment by milesial** [comment]
You're right! Sorry about that, I did the recent refactor a bit in a rush..
Fixed in master now, thanks for spotting this.
